### DISCUSSION TITLE
Bluetooth Keyboard and Mouse Incorrectly Assigned as Controllers

### DISCUSSION THREAD

**Discussion by Mittens0407** [darthcloud/BlueRetro#676] [Q&A] [answered]
Hello. I've been attempting to use my Bluetooth keyboard and mouse (https://a.co/d/cvzohzj) with the Dreamcast BlueRetro adapter from Retro Scaler (https://www.aliexpress.us/item/3256805176616098.html). They both manage to pair, but according to 240p Test Suite and all games I've tested, they are being detected as controllers and not as mouse and keyboard peripherals. Interestingly, the keyboard has some buttons mapped to it, which led me to incorrectly assume that it was working the first time I tested it. The mouse doesn't do anything however. I have flashed the latest firmware onto it from the recommended repository (https://github.com/RetroScaler/DC-Blueretro/releases), and I believe it's in line with the latest firmware here. Any help would be appreciated.

**Comment by darthcloud** [comment, marked as answer]
You need to set in the output config the device you want BlueRetro to emulate:

https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-BLE-Web-Config-User-Manual#23---output-config
